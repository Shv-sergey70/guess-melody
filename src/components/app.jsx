import React from 'react';
import WelcomeScreen from './welcome-screen';

const App = () => {
  return (
    <WelcomeScreen/>
  );
};

export default App;
