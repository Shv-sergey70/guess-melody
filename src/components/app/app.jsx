import React from 'react';
import PropTypes from 'prop-types';
import WelcomeScreen from '../welcome-screen/welcome-screen';
import ArtistQuestionScreen from "../artist-question-screen/artist-question-screen";
import GenreQuestionScreen from "../genre-question-screen/genre-question-screen";
import {connect} from "react-redux";

const App = ({currentStep, questions, time, attempts}) => {
  if (currentStep === -1) {
    return (
      <WelcomeScreen time={time} attempts={attempts} />
    );
  }

  switch (questions[currentStep].type) {
    case `genre`:
      return (
        <GenreQuestionScreen
          question={questions[currentStep]}
          attempts={attempts} />
      );
    case `artist`:
      return (
        <ArtistQuestionScreen
          question={questions[currentStep]}
          attempts={attempts} />
      );
  }

  return null;
};

App.propTypes = {
  questions: PropTypes.arrayOf(PropTypes.object),
  time: PropTypes.number.isRequired,
  attempts: PropTypes.number.isRequired,
  currentStep: PropTypes.number.isRequired
};

const mapStateToProps = ({step}) => ({
  currentStep: step
});

export {App};
export default connect(mapStateToProps)(App);
