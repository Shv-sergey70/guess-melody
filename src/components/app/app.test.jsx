import React from 'react';
import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {App} from "./app";

configure({adapter: new Adapter()});

const questions = [
  {
    type: `genre`,
    genre: `rock`,
    answers: [
      {
        src: `./tracks/Anivar_Summer.mp3`,
        genre: `pop`
      },
      {
        src: `./tracks/Billie_Eilish_ilomilo.mp3`,
        genre: `jazz`
      },
      {
        src: `./tracks/Cj_L_Black_Rain.mp3`,
        genre: `rock`
      },
      {
        src: `./tracks/Orquesta_L_rica_de_Barcelona_Las_Cuatro_Estaciones,_Invierno_Allegro_Non_Molto.mp3`,
        genre: `classic`
      }
    ]
  },
  {
    type: `artist`,
    song: {
      artist: `Artur Latte`,
      src: `./tracks/Raim_Artur_Latte.mp3`,
    },
    answers: [
      {
        picture: `/img/default-artist.png`,
        artist: `Billie Eilish`
      },
      {
        picture: `/img/default-artist.png`,
        artist: `Imagine Dragons`
      },
      {
        picture: `/img/default-artist.png`,
        artist: `Artur Latte`
      }
    ]
  }
];

describe(`App correctly renders`, () => {
  test(`Welcome screen`, () => {
    const tree = shallow(
        <App
          time={300}
          attempts={3}
          questions={questions}
          currentStep={-1}
          onAnswer={jest.fn()} />
    );

    expect(tree).toMatchSnapshot();
  });

  test(`Genre-question screen`, () => {
    const tree = shallow(
        <App
          time={300}
          attempts={3}
          questions={questions}
          currentStep={0}
          onAnswer={jest.fn()} />
    );

    expect(tree).toMatchSnapshot();
  });

  test(`Artist-question screen`, () => {
    const tree = shallow(
        <App
          time={300}
          attempts={3}
          questions={questions}
          currentStep={1}
          onAnswer={jest.fn()} />
    );

    expect(tree).toMatchSnapshot();
  });

  test(`Losing screen`, () => {
    const tree = shallow(
        <App
          time={0}
          attempts={3}
          questions={questions}
          currentStep={1}
          onAnswer={jest.fn()} />
    );

    expect(tree).toMatchSnapshot();
  });
});
