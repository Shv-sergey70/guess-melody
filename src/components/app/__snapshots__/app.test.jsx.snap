// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App correctly renders Artist-question screen 1`] = `
<WithActivePlayer
  question={
    Object {
      "answers": Array [
        Object {
          "artist": "Billie Eilish",
          "picture": "/img/default-artist.png",
        },
        Object {
          "artist": "Imagine Dragons",
          "picture": "/img/default-artist.png",
        },
        Object {
          "artist": "Artur Latte",
          "picture": "/img/default-artist.png",
        },
      ],
      "song": Object {
        "artist": "Artur Latte",
        "src": "./tracks/Raim_Artur_Latte.mp3",
      },
      "type": "artist",
    }
  }
/>
`;

exports[`App correctly renders Attempts losing screen 1`] = `
<Connect(LosingScreen)>
  <h2
    className="result__title"
  >
    Какая жалость!
  </h2>
  <p
    className="result__total result__total--fail"
  >
    У вас закончились все попытки. Ничего, повезёт в следующий раз!
  </p>
</Connect(LosingScreen)>
`;

exports[`App correctly renders Authorization is required 1`] = `<WithLogin />`;

exports[`App correctly renders Genre-question screen 1`] = `
<WithUserAnswers
  onAnswer={[MockFunction]}
  question={
    Object {
      "answers": Array [
        Object {
          "genre": "pop",
          "src": "./tracks/Anivar_Summer.mp3",
        },
        Object {
          "genre": "jazz",
          "src": "./tracks/Billie_Eilish_ilomilo.mp3",
        },
        Object {
          "genre": "rock",
          "src": "./tracks/Cj_L_Black_Rain.mp3",
        },
        Object {
          "genre": "classic",
          "src": "./tracks/Orquesta_L_rica_de_Barcelona_Las_Cuatro_Estaciones,_Invierno_Allegro_Non_Molto.mp3",
        },
      ],
      "genre": "rock",
      "type": "genre",
    }
  }
/>
`;

exports[`App correctly renders Time losing screen 1`] = `
<Connect(LosingScreen)>
  <h2
    className="result__title"
  >
    Увы и ах!
  </h2>
  <p
    className="result__total result__total--fail"
  >
    Время вышло! Вы не успели отгадать все мелодии
  </p>
</Connect(LosingScreen)>
`;

exports[`App correctly renders Welcome screen 1`] = `
<Connect(WelcomeScreen)
  attempts={3}
/>
`;
